exchange/exchange.go

```diff
@@ -348,13 +348,12 @@ func (e *exchange) HoldAuction(ctx context.Context, r *AuctionRequest, debugLog
 		liveAdapters []openrtb_ext.BidderName
 	)
 
-	if len(r.StoredAuctionResponses) > 0 {
-		adapterBids, fledge, liveAdapters, err = buildStoredAuctionResponse(r.StoredAuctionResponses)
+	shouldMspBackfillBids := mspUpdateStoredAuctionResponse(r)
+	if len(r.StoredAuctionResponses) > 0 && !shouldMspBackfillBids {
+		adapterBids, fledge, liveAdapters, err, anyBidsReturned = mspApplyStoredAuctionResponse(r)
```

Comment:
For in-house ads feature, the best way to do it is setting up an endpoint and creating a bid adapter. If you think this way is over complicated for now, then `ExecuteAuctionResponseStage` is an alternative approach. We should avoid updating Prebid-server code as much as possible.

Reply:
we need to pass more parameters for ExecuteAuctionResponseStage approach and reprocess line 384 to 505